assets/js/DataGrid.js

```diff
@@ -94,6 +94,90 @@ class ReorderColumnDropZone extends React.PureComponent {
   }
 }
 
+export class EditableColumnName extends React.Component {
+  static propTypes = {
+    columnKey: PropTypes.string.isRequired,
+    onRename: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newName: props.columnKey,
+      editMode: false,
+    };
+
+    this.enterEditMode = this.enterEditMode.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleInputBlur = this.handleInputBlur.bind(this);
+    this.handleInputKeyPress = this.handleInputKeyPress.bind(this);
+    this.handleInputFocus = this.handleInputFocus.bind(this);
+  }
+
+  enterEditMode() {
+    this.setState({editMode: true});
+  }
+
+  handleInputChange(event) {
+    this.setState({newName: event.target.value});
+  }
+
+  handleInputCommit() {
+    this.setState({
+        newName: this.state.newName,
+        editMode: false
+    });
+    if(this.state.newName != this.props.columnKey) {
+      this.props.onRename({
+        prevName: this.props.columnKey,
+        newName: this.state.newName
+      });
+    }
+  }
+
+  handleInputBlur() {
+    this.handleInputCommit();
+  };
+
+  handleInputKeyPress(event) {
+    if(event.key == 'Enter') {
+      this.handleInputCommit();
+    }
```

Comment:
I think we should handle 'Escape' here: just do `setState()` back to the input props.